src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -70,6 +70,9 @@
   private final AtomicBoolean running = new AtomicBoolean(false);
   private final AtomicLong taskThreadId = new AtomicLong(0);
 
+  private Map<TableQuerier, Integer> retriesAttemptedPerTableQuerier;
```

Comment:
I see that we're using a similar strategy for the counting the consecutiveEmptyResults, but that's a local map only used within a single poll.

WDYT about making this a mutable state field inside the TableQuerier instead of a map? This sort of "external member field" seems to break some OOP rules.

Reply:
yeah sounds good to me.

Reply:
I don't think you need that if the state is mutable via a method that signals the exception and asks the querier to  increment the retry counter. @gharris1727 is that what you had in mind?